### PR TITLE
ci: add workflow to close stale issues

### PR DESCRIPTION
**Pull request by maschad**
Closes https://github.com/FuelLabs/fuels-ts/issues/2203

Introduces a workflow to clean up potentially stale issues / bugs raised where we need author input to progress.

This workflow requires two new labels `need-author-input` and `stale`

So the sequence of actions would be this:

1. A user (whether code owner or not) creates an issue, no label is assigned to it.
2. A code owner responds to that issue, but determines that more information is needed to proceed, they assign the `need-author-input` label
3. Only one that `need-author-input` label has been assigned to the issue, the countdown (of 14 days in this case) begins, after 14 days the label stale is assigned to the issue and a generated warning is commented by the github bot that the issue will close in 7 days
4. After 7 days of the issue being stale it is closed by the github bot.

### CODE REVIEW

File: .github/workflows/stale.yml
Comment (maschad) [reply .github/workflows/stale.yml]: definitely not 😅

File: .github/workflows/stale.yml
Comment (arboleya) [inline .github/workflows/stale.yml:16]: `github.token`?

File: .github/workflows/stale.yml
Comment (arboleya) [inline .github/workflows/stale.yml:23]: IIUC, statistics depend on [enabling logs](https://github.com/actions/stale?tab=readme-ov-file#debugging) - is it worth it?

Comment (maschad) [reply .github/workflows/stale.yml:23]: probably not at this time, we can re-visit in the future

File: .github/workflows/stale.yml
Comment (maschad) [reply .github/workflows/stale.yml:16]: copy pasta 🤪

File: .github/workflows/stale.yml
Comment (arboleya) [inline .github/workflows/stale.yml:21]: Is 6x days enough to mark things as stale?

Could you please clarify the criteria by which it happens?

For example, will it apply only to issues created by non-code owners?

Comment (danielbate) [reply .github/workflows/stale.yml:21]: Could you add this to the PR description?

**Review by danielbate — APPROVED** [review]

File: .github/workflows/stale.yml
Comment (arboleya) [inline .github/workflows/stale.yml:20]: We should use a shorter version more aligned with the other labels.

```suggestion
          any-of-labels: "awaiting"
```

**Review by arboleya — CHANGES_REQUESTED** [review]

File: .github/workflows/stale.yml
Comment (arboleya) [reply .github/workflows/stale.yml:21]: @maschad Perfect, thanks for clarifying.

I think 6x days may not be enough, can we start with 14 days and see how it goes?

Comment (maschad) [reply .github/workflows/stale.yml:21]: Sure

File: .github/workflows/stale.yml
Comment (maschad) [reply .github/workflows/stale.yml:20]: I can appreciate the brevity but I think _awaiting_ is a bit ambiguous, I would suggest _awaiting-input_ but that's just two characters shorter than _need-author-input_

Comment (arboleya) [reply .github/workflows/stale.yml:20]: Apologies, I expressed myself poorly. I was referring to the label's simplicity (single word, no hyphens). I believe it's a no-brainer for us to understand what it means, and we're the only ones who actually need to know.

A question like `Awaiting for what?` has arguably the same weight as `Blocked by what?`. It doesn't seem like a problem to me, but if it becomes one, we could disambiguate in the label's description if necessary.

Either way, it's worth considering its use will be marginal right now, another reason I'd favor the simplicity of words and broad terms. If external participation increases and it becomes strictly necessary (as in, a _problem_), we can always revisit and rename it. 🙂

Comment (maschad) [reply .github/workflows/stale.yml:20]: I take your point

**Review by arboleya — APPROVED** [review]
Thanks for the PR and adjustments! 🙌

**Review by Torres-ssf — APPROVED** [review]
🚀

**Pull request merged**
